flang/lib/Optimizer/Passes/Pipelines.cpp

```diff
@@ -350,15 +350,11 @@ void createDefaultFIRCodeGenPassPipeline(mlir::PassManager &pm,
   else
     framePointerKind = mlir::LLVM::framePointerKind::FramePointerKind::None;
 
-  bool setNoCapture = false, setNoAlias = false;
-  if (config.OptLevel.isOptimizingForSpeed())
```

Comment:
Tom, will you be okay if I just remove this `if`?  The LIT tests should still work, and it will be easier to re-enable the code, when function speciailization is fixed.